# Video Games

## *Which company is the most common video game publisher?*

In [1]:
import pandas as pd
import numpy as np
import datetime as dt

In [2]:
df = pd.read_csv('./vgsales.csv')


In [3]:
df.head()


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [27]:
sales = df.iloc[:,6:11]
df = sales > 70
sales.tail()


NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
16593      0.01      0.00       0.0          0.0          0.01
16594      0.01      0.00       0.0          0.0          0.01
16595      0.00      0.00       0.0          0.0          0.01
16596      0.00      0.01       0.0          0.0          0.01
16597      0.01      0.00       0.0          0.0          0.01

In [31]:
# df[ (df.column0 == thing1)
df(df.Global_Sales >= .1)
# df[df['Other_Sales']>.1]
# df[df['JP_Sales']>.1]
# df[df['EU_Sales']>.1]
# df[df['NA_Sales']>.2]

df.head()





TypeError: 'DataFrame' object is not callable

In [5]:
df.Publisher

0                           Nintendo
1                           Nintendo
2                           Nintendo
3                           Nintendo
4                           Nintendo
5                           Nintendo
6                           Nintendo
7                           Nintendo
8                           Nintendo
9                           Nintendo
10                          Nintendo
11                          Nintendo
12                          Nintendo
13                          Nintendo
14                          Nintendo
15            Microsoft Game Studios
16              Take-Two Interactive
17              Take-Two Interactive
18                          Nintendo
19                          Nintendo
20                          Nintendo
21                          Nintendo
22                          Nintendo
23              Take-Two Interactive
24              Take-Two Interactive
25                          Nintendo
26                          Nintendo
2

In [6]:

df.Publisher.value_counts().head(10)

Electronic Arts                 1351
Activision                       975
Namco Bandai Games               932
Ubisoft                          921
Konami Digital Entertainment     832
THQ                              715
Nintendo                         703
Sony Computer Entertainment      683
Sega                             639
Take-Two Interactive             413
Name: Publisher, dtype: int64

 #  The most common video game publisher is Electronic Arts  with 1351 titles. 

## *What’s the most common platform?*

In [7]:
df.head()


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [8]:
df.Platform.value_counts().head(10)


DS      2163
PS2     2161
PS3     1329
Wii     1325
X360    1265
PSP     1213
PS      1196
PC       960
XB       824
GBA      822
Name: Platform, dtype: int64

# The DS is the most common platform with 2163 titles


## *What about the most common genre?*

In [9]:
df.Genre.value_counts()


Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: Genre, dtype: int64

# Action is the most common genre with 3316 titles

## *What are the top 20 highest grossing games?*

In [10]:
descend= df.sort_values(by='Global_Sales', ascending=0)

In [11]:
descend.head(20)

Rank                                          Name Platform    Year  \
0      1                                    Wii Sports      Wii  2006.0   
1      2                             Super Mario Bros.      NES  1985.0   
2      3                                Mario Kart Wii      Wii  2008.0   
3      4                             Wii Sports Resort      Wii  2009.0   
4      5                      Pokemon Red/Pokemon Blue       GB  1996.0   
5      6                                        Tetris       GB  1989.0   
6      7                         New Super Mario Bros.       DS  2006.0   
7      8                                      Wii Play      Wii  2006.0   
8      9                     New Super Mario Bros. Wii      Wii  2009.0   
9     10                                     Duck Hunt      NES  1984.0   
10    11                                    Nintendogs       DS  2005.0   
11    12                                 Mario Kart DS       DS  2005.0   
12    13                   Pokemon Gold/Pokemon Silver       GB  1999.0   
13    14                                       Wii Fit      Wii  2007.0   
14    15                                  Wii Fit Plus      Wii  2009.0   
15    16                            Kinect Adventures!     X360  2010.0   
16    17                            Grand Theft Auto V      PS3  2013.0   
17    18                 Grand Theft Auto: San Andreas      PS2  2004.0   
18    19                             Super Mario World     SNES  1990.0   
19    20  Brain Age: Train Your Brain in Minutes a Day       DS  2005.0   

           Genre               Publisher  NA_Sales  EU_Sales  JP_Sales  \
0         Sports                Nintendo     41.49     29.02      3.77   
1       Platform                Nintendo     29.08      3.58      6.81   
2         Racing                Nintendo     15.85     12.88      3.79   
3         Sports                Nintendo     15.75     11.01      3.28   
4   Role-Playing                Nintendo     11.27      8.89     10.22   
5         Puzzle                Nintendo     23.20      2.26      4.22   
6       Platform                Nintendo     11.38      9.23      6.50   
7           Misc                Nintendo     14.03      9.20      2.93   
8       Platform                Nintendo     14.59      7.06      4.70   
9        Shooter                Nintendo     26.93      0.63      0.28   
10    Simulation                Nintendo      9.07     11.00      1.93   
11        Racing                Nintendo      9.81      7.57      4.13   
12  Role-Playing                Nintendo      9.00      6.18      7.20   
13        Sports                Nintendo      8.94      8.03      3.60   
14        Sports                Nintendo      9.09      8.59      2.53   
15          Misc  Microsoft Game Studios     14.97      4.94      0.24   
16        Action    Take-Two Interactive      7.01      9.27      0.97   
17        Action    Take-Two Interactive      9.43      0.40      0.41   
18      Platform                Nintendo     12.78      3.75      3.54   
19          Misc                Nintendo      4.75      9.26      4.16   

    Other_Sales  Global_Sales  
0          8.46         82.74  
1          0.77         40.24  
2          3.31         35.82  
3          2.96         33.00  
4          1.00         31.37  
5          0.58         30.26  
6          2.90         30.01  
7          2.85         29.02  
8          2.26         28.62  
9          0.47         28.31  
10         2.75         24.76  
11         1.92         23.42  
12         0.71         23.10  
13         2.15         22.72  
14         1.79         22.00  
15         1.67         21.82  
16         4.14         21.40  
17        10.57         20.81  
18         0.55         20.61  
19         2.05         20.22

# The above chart shows the top 20 grossing video games

## *For North American video game sales, what’s the median?*

In [12]:
df.head()


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [13]:
df.NA_Sales.median()

0.08

# The median sales for North America is .08 (million) 




##  *Provide a secondary output showing ‘about’ ten games surrounding the median sales output*